lib/vmdb/settings/database_source.rb

```diff
@@ -3,23 +3,57 @@ class Settings
     class DatabaseSource
       include Vmdb::Logging
 
-      attr_reader :resource
+      attr_reader :resource_instance, :settings_holder_class
 
-      def initialize(resource)
-        @resource = resource
+      # NOTE: resource - instance of MiqRegion, Zone or MiqServer
+      #       klass    - class name of object which provides access to SettingsChange for resource
+      #                  possible values of klass are in SETTINGS_HIERARCHY
+      # EXAMPLE: 1. resource is instance of MiqServer and klass is 'Zone',
+      #             than settings will be loaded from resource.zone.settings_changes
+      #          2. resource is instance of MiqServer and klass is 'MiqRegion'
+      #             than settings will be loaded from resource.miq_region.settings_changes
+      
+      def initialize(resource, klass)
```

Comment:
Minor but prefer yardoc [[1]](http://yardoc.org/) [[2]](http://www.rubydoc.info/gems/yard/file/docs/GettingStarted.md) for documenting.   Also, the comment should be right up against the initialize...that would make this:

```ruby
# @param resource [MiqRegion, Zone, MiqServer] the resource
# @param klass [Class] class name of object which provides access to SettingsChange for resource
#   possible values of klass are {DatabaseSources::SETTINGS_HIERARCHY the settings hierarchy}
#
#   Example:
#   1. resource is instance of `MiqServer` and klass is 'Zone', then settings will be loaded from `resource.zone.settings_changes`
#   2. resource is instance of `MiqServer` and klass is 'MiqRegion', then settings will be loaded from `resource.miq_region.settings_changes`
def initialize(resource, klass)
```

You might also be able to use `@example` tag, but I'm not sure.